### PR TITLE
3257 Show documents properly in the "Only show PDFs" query

### PR DESCRIPTION
**Pull request by ERosendo**
This PR fixes the full join query by adding children filters to the parent query when using the available_only field.

The filter added in #3301 is intended to return only dockets with available documents that match the search query, it works properly when used independently or in conjunction with parent filters( like the case name) but It's not specific enough and when it's combined with child filters (such as the description filter), it also returns dockets that have a document available but don't match the search criteria.

### CODE REVIEW

**Review by albertisfu — COMMENTED** [review]
Thanks, yeah this seems good to me! 

I just left a comment suggesting that we should modify the code to pass the `has_child` query as a filter for parent documents whenever any child filter is applied. Excluding the parties and attorneys filters, as they are not part of the RECAPDocument mapping. I noted this because the issue seems to affect other child filters, not just the `available_only` filter.

Additionally, if we remove the parties and attorneys filters from the `has_child` query, it will resolve an issue I discovered reviewing this one. Specifically, if you use either the party or attorney filter, it will exclude all dockets with child documents even if the docket actually contains the party specified in the filter. This occurs because the parties' fields are not a part of the RECAPDocument.

For instance, a query like this:
https://www.courtlistener.com/?q=Voluntary%20hospital&type=r&order_by=score%20desc&party_name=Frank%20Paul%20Sabatini

Will return no results due to this issue.

After we remove the parties and attorneys fields from the `has_child` query, the previous query will return cases with or without child docs that either match the text query, the filter, or cases that contain the specified attorney or party. However, we won't be able to filter child documents using a party filter. This is part of the trade-off we discussed when we decided to only index parties into the DocketDocument.

File: cl/search/tests/tests_es_recap.py [hunk 1]
```diff
@@ -573,7 +573,128 @@ def test_available_only_field(self) -> None:
...
+            "type": SEARCH_TYPES.RECAP,
+            "case_name": "Voluntary",
+            "description": "Voluntary",
+            "available_only": True,
+        }
+        r = async_to_sync(self._test_article_count)(
+            params, 1, "case_name + available_only"
+        )
+        self.assertIn("Document #1", r.content.decode())
+
```
[the remark this answers is not in the document]
Comment (ERosendo) [reply cl/search/tests/tests_es_recap.py:728]: Thank you for your suggestion. I will add these test cases.

File: cl/lib/elasticsearch_utils.py [hunk 2]
```diff
@@ -1528,12 +1528,17 @@ def build_full_join_es_queries(
 
         # Adds filter to the parent query to exclude results with no children
         if cd.get("available_only", ""):
+            query = (
+                Q("bool", filter=child_filters)
+                if child_filters
+                else Q("term", is_available=True)
+            )
```
[the remark this answers is not in the document]
Comment (ERosendo) [reply cl/lib/elasticsearch_utils.py]: You're right about the refactor. Thanks!

I'll apply your suggestion and also tweak the logic to create the `child_filters`.

**Comment by ERosendo** [comment]
@albertisfu Thanks for reviewing this PR. I've applied your suggestions to refactor the code and fix the parties/attorneys filter issue.

**Review by albertisfu — APPROVED** [review]
Great, this looks good to me, thanks!

**Pull request merged**